pyrefly/lib/test/lsp/lsp_interaction/diagnostic.rs

```diff
@@ -163,3 +166,87 @@ fn test_error_documentation_links() {
 
     interaction.shutdown();
 }
+
+#[test]
+fn test_version_support_publish_diagnostics() {
+    let test_files_root = get_test_files_root();
+    let root = test_files_root.path().to_path_buf();
+    let mut file = root.clone();
+    file.push("text_document.py");
+    let uri = Url::from_file_path(file).unwrap();
+    let mut interaction = LspInteraction::new();
+    interaction.set_root(root);
+    interaction.initialize(InitializeSettings {
+        configuration: Some(None),
+        capabilities: Some(serde_json::json!({
+            "textDocument": {
+                "publishDiagnostics": {
+                    "versionSupport": true,
+                },
+            },
+        })),
+        ..Default::default()
+    });
+
+    interaction.server.did_open("text_document.py");
+    interaction.server.diagnostic("text_document.py");
+
+    interaction
+        .client
+        .expect_message(lsp_server::Message::Notification(
+            lsp_server::Notification {
+                method: "textDocument/publishDiagnostics".to_owned(),
+                params: serde_json::json! {{
+                    "uri": uri,
+                    "diagnostics": [],
+                    "version": 1
+                }},
+            },
+        ));
+
+    interaction.server.did_change("text_document.py", "# test");
+    interaction.server.diagnostic("text_document.py");
+
+    // I don't understand why this version is still 1
+    interaction
+        .client
+        .expect_message(lsp_server::Message::Notification(
+            lsp_server::Notification {
+                method: "textDocument/publishDiagnostics".to_owned(),
```

Comment:
if we send a server.diagnostic request we should receive a [document diagnostic](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_diagnostic) response (maybe we also get a publishDiagnostic)